src/_pytest/nodes.py

```diff
@@ -109,45 +110,36 @@ class Node(metaclass=NodeMeta):
         "parent",
         "config",
         "session",
-        "fspath",
+        "fs_path",
         "_nodeid",
         "_store",
         "__dict__",
     )
 
     def __init__(
         self,
+        *,
         name: str,
-        parent: "Optional[Node]" = None,
-        config: Optional[Config] = None,
-        session: "Optional[Session]" = None,
-        fspath: Optional[py.path.local] = None,
-        nodeid: Optional[str] = None,
+        parent: Optional["Node"],
+        config: Config,
+        session: "Session",
+        fs_path: Path,
+        nodeid: str,
     ) -> None:
         #: A unique name within the scope of the parent node.
         self.name = name
-
+        self._nodeid = nodeid
         #: The parent collector node.
         self.parent = parent
 
         #: The pytest config object.
-        if config:
-            self.config: Config = config
-        else:
-            if not parent:
-                raise TypeError("config or parent must be provided")
-            self.config = parent.config
+        self.config = config
 
         #: The pytest session this node is part of.
-        if session:
-            self.session = session
-        else:
-            if not parent:
-                raise TypeError("session or parent must be provided")
-            self.session = parent.session
+        self.session = session
 
         #: Filesystem path where this node was collected from (can be None).
-        self.fspath = fspath or getattr(parent, "fspath", None)
+        self.fs_path = fs_path
```

Comment:
Being a bit nitpicky, but `fs_path` and `fspath` are too close to one another, I fear users will get the two confused all the time.

How about: `path` for the `Path` version? After all the `fs` prefix is a bit redudant.

Reply:
I agree with @nicoddemus, the name `fs_path` is not very pretty.

One thing against just `path` is that someday we'll also want to migrate our hooks to `Path` and their the py.path param is always `path` so the Path param would need a different name. And `path` is not very distinguishable either. But I can't think of something else.

Reply:
How about `file_path`? This is always a file right?

Reply:
there are situations where it ought to be a folder
* external collection utilities
* collection roots (a concept i intend to introduce as extension of the testpaths configuration (i think it needs a ticket)
* Packages ought to use it as well